### DISCUSSION TITLE
Can I call `then_order_by` before `order_by`?

### DISCUSSION THREAD

**Discussion by Zizico2** [diesel-rs/diesel#3481] [Q&A] [answered]
The examples in the docs always have `order_by` as the first call, and then `then_order_by`. This can be quite cumbersome to achieve inside loops though.

**Comment by weiznich** [comment, marked as answer]
If there is no existing `order` clause both calls are equivalent and will generate the same SQL. That means you can call `.then_order_by` unconditionally in your loop.
